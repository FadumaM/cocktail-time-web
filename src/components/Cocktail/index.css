.cocktail-container {
  height: 80%;
  display: flex;
  width: 100%;
  padding: 3em;
}

.cocktail-image,
.cocktail-details {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.cocktail-image,
.cocktail-details {
  width: 50%;
}
.cocktail-details {
  padding: 3em;
}
.cocktail-details p {
  max-width: 400px;
  word-wrap: break-word;
}

.cocktail-image img {
  width: 100%;
  object-fit: scale-down;
  border-radius: 0.3em;
}

ol.ingredient {
  padding-inline-start: 1em;
}

ol.ingredient li {
  margin-bottom: 1em;
  font-weight: bold;
}

ol.ingredient li span {
  font-weight: normal;
}

@media only screen and (max-width: 600px) {
  .cocktail {
    height: auto;
    margin: 3em auto;
    width: 90%;
  }
  .cocktail-container {
    height: auto;
    flex-direction: column;
    padding: 0;
  }
  .cocktail-image,
  .cocktail-details {
    width: 100%;
    padding: 0;
  }
}
