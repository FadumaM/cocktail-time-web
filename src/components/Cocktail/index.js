import React from "react";
import "./styles.css";

const Cocktail = ({ cocktail }) =>
  cocktail && Object.entries(cocktail).length > 0 ? (
    <div className="cocktail-container">
      <div className="cocktail-image">
        <img src={cocktail.image} alt={cocktail.name} />
      </div>
      <div className="cocktail-details">
        <h1>{cocktail.name}</h1>
        <div className="cocktail-instruction">
          <h2>Instruction</h2>
          <p>{cocktail.instruction}</p>
        </div>
        <div className="cocktail-ingredients">
          <h3>Ingredients</h3>
          <ol className="ingredient">
            {cocktail.ingredients.map(({ amount, name, id }, index) => (
              <li key={index}>
                <span>
                  {amount} {name}
                </span>
              </li>
            ))}
          </ol>
        </div>
      </div>
    </div>
  ) : null;

export default Cocktail;
