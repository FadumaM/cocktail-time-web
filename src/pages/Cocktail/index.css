.random,
.cocktail {
  width: 90%;
  margin: 0 auto;
  display: flex;
  height: 90vh;
  justify-content: center;
  align-items: center;
}
