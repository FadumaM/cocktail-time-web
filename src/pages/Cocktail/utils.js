import { gql } from "apollo-boost";

export const GET_COCKTAIL = gql`
  query GetCocktail($id: ID!) {
    getCocktail(id: $id) {
      id
      name
      ingredients {
        amount
        name
      }
      image
      instruction
    }
  }
`;
