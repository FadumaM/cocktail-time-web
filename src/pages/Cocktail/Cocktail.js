import React from "react";
import "./Cocktail.css";

const Cocktail = ({ cocktail }) =>
  console.log(cocktail) || (
    <div className="cocktail">
      <div className="cocktail-container">
        <div className="cocktail-image">
          <img src={cocktail.image} alt={cocktail.name} />
        </div>
        <div className="cocktail-details">
          <h1>{cocktail.name}</h1>
          <div className="cocktail-instruction">
            <h3>Instruction</h3>
            <p>{cocktail.instruction}</p>
          </div>
          <div className="cocktail-ingredients">
            <h3>Ingredients</h3>
            <ol className="ingredient">
              {cocktail.ingredients.map(({ amount, name, id }, index) => (
                <li key={index}>
                  <span>
                    {amount} {name}
                  </span>
                </li>
              ))}
            </ol>
          </div>
        </div>
      </div>
    </div>
  );

export default Cocktail;
