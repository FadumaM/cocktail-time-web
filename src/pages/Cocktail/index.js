import React from "react";
import { useQuery } from "@apollo/react-hooks";
import { useParams, useLocation } from "react-router-dom";
import { GET_COCKTAIL, GET_RANDOM_COCKTAIL } from "./utils";
import Cocktail from "../../components/Cocktail";

const WithCocktailId = (Component) => {
  const { id } = useParams();
  const { data, loading } = useQuery(GET_COCKTAIL, {
    variables: { id },
  });

  if (loading) {
    return null;
  }
  return (
    <div data-testid="cocktail-page">
      <Component cocktail={data.getCocktailById} />
    </div>
  );
};
const WithRandomCocktail = (Component) => {
  const { data, loading } = useQuery(GET_RANDOM_COCKTAIL);

  if (loading) {
    return null;
  }
  return (
    <div data-testid="random-cocktail-page">
      <Component cocktail={data.getRandomCocktail} />
    </div>
  );
};

export default () => {
  const { pathname } = useLocation();
  return pathname === "/random"
    ? WithRandomCocktail(Cocktail)
    : WithCocktailId(Cocktail);
};
