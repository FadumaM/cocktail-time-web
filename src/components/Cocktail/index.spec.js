import React from "react";
import { render } from "@testing-library/react";
import Cocktail from "./index";
describe("Cocktail", () => {
  let mockCocktail;
  beforeEach(() => {
    mockCocktail = {
      id: "1",
      name: "Cocktail name",
      instruction: "Cocktail instruction",
      ingredients: [
        { name: "rum", amount: "1 oz" },
        { name: "lime", amount: "1" },
      ],
    };
  });

  it("should render correctly", () => {
    const { baseElement } = render(<Cocktail cocktail={mockCocktail} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should render correct amount of ingredients", () => {
    const { container } = render(<Cocktail cocktail={mockCocktail} />);
    const ingredients = container.querySelectorAll("li");
    expect(ingredients).toHaveLength(2);
  });
});
