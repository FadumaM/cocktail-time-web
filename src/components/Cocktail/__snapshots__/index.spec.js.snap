// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail should render correctly 1`] = `
<body>
  <div>
    <div
      class="cocktail-container"
    >
      <div
        class="cocktail-image"
      >
        <img
          alt="Cocktail name"
        />
      </div>
      <div
        class="cocktail-details"
      >
        <h1>
          Cocktail name
        </h1>
        <div
          class="cocktail-instruction"
        >
          <h3>
            Instruction
          </h3>
          <p>
            Cocktail instruction
          </p>
        </div>
        <div
          class="cocktail-ingredients"
        >
          <h3>
            Ingredients
          </h3>
          <ol
            class="ingredient"
          >
            <li>
              <span>
                1 oz
                 
                rum
              </span>
            </li>
            <li>
              <span>
                1
                 
                lime
              </span>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
</body>
`;
